src/sqlfluff/core/rules/analysis/select.py

```diff
@@ -132,6 +132,7 @@ def _has_value_table_function(table_expr, dialect):
         return False
 
     for function_name in table_expr.recursive_crawl("function_name"):
-        if function_name.raw.lower() in dialect.sets("value_table_functions"):
+        # Other rules can increase whitespace in the function name so use strip
+        if function_name.raw.lower().strip() in dialect.sets("value_table_functions"):
```

Comment:
I saw your explanation of this fix on the related issue. I think that behavior (rule adding whitespace to function names) is a deeper bug that we should investigate and fix as well. The correct behavior would be to add the whitespace as a new segment (or possibly edit/extend existing whitespace to the left).

We seem to discover these kinds of rule bugs every month or two: a rule that is doing something wrong, but it doesn't cause an obvious problem most of the time.

I have this idea that SQLFluff should have various runtime checks on the `LintFix` objects returned by rules to prevent them from misbehaving. I should probably create a ticket for that. A "linter for the linter", kind of.

Reply:
In fact, I think if we fix that deeper issue, this change will no longer be needed, if that makes sense. Even during the "fix" phase, segments should always correspond to something the parser and the dialect would've created, but in this case, that's not true (whitespace and code in the same segment).

Reply:
I don't disagree.

However I still think we should accept this fix in the meantime because:
1. This is a nasty little bug that breaks SQL. I think it's pretty common to have no indentation on `UNNEST` statements - it certainly is in our code base and don't think we're unusually in that regard - and running `sqlfluff fix` on any such code will break it.
2. The fix is tiny! And there's no real downsides to it being in there (the overhead of a `strip()` of a text which is just space and a function name is negligible!) so even though we could removed it if the larger fix is done, there's no downside to leaving it in place.
3. I have now added the test case, which will be useful for when we do the get round to the bigger fix.
4. The bigger fix might be quite large! Presumably the space is there to allow rule 10 to correct indentation. So that whole logic may have to be rewritten. This could also impact all the other rules that do indentation (and there's lots of them).
 
I can't commit to being able to do this rewrite any time soon and presume no one else can either. So my big preference is to merge this and open an issue for the bigger fix if you agree.

And btw this is such a nasty little bug (that broke lots of our queries!), that I was going to push for a 0.6.4 (even though we only released 0.6.3 fairly recently) once merged.

Reply:
That sounds reasonable. Could you go ahead and open that other issue, then add a comment to the line above to explain why the `.strip()` is there? Once someone fixes the other issue, it will be cool to see if we can remove the above line without breaking the test.

Reply:
Done!